lwgenerate/azure/azure.go

```diff
@@ -0,0 +1,460 @@
+
+import (
+	"github.com/hashicorp/hcl/v2/hclwrite"
+	"github.com/lacework/go-sdk/lwgenerate"
+	"github.com/pkg/errors"
+)
+
+type GenerateAzureTfConfigurationArgs struct {
+	// Should we configure Activity Log integration in LW?
+	ActivityLog bool
+
+	// Should we add Config integration in LW?
+	Config bool
+
+	// Should we create an Active Directory integration
+	AdIntegration bool
+
+	// If Config is true, give the user the opportunity to name their integration. Defaults to "TF Config"
+	ConfigIntegrationName string
+
+	// If ActivityLog is true, give the user the opportunity to name their integration. Defaults to "TF activity log"
+	ActivityLogIntegrationName string
+
+	// Active Directory application Id
+	AdApplicationId string
+
+	// Active Directory password
+	AdApplicationPassword string
+
+	// Active Directory Enterprise app object id
+	AdServicePrincipalId string
+
+	// Should we use the management group, rather than subscription
+	ManagementGroup bool
+
+	// Management Group ID to set
+	ManagementGroupId string
+
+	// List of subscription Ids
+	SubscriptionIds []string
+
+	// Grant read access to ALL subscriptions
+	AllSubscriptions bool
+
+	// Storage Account name
+	StorageAccountName string
+
+	// Storage Account Resource Group
+	StorageAccountResourceGroup string
+
+	// Should we use existing storage account
+	ExistingStorageAccount bool
+
+	// Azure region where the storage account for loggin resides
+	Location string
+}
+
+// Ensure all combinations of inputs are valid for supported spec
+func (args *GenerateAzureTfConfigurationArgs) validate() error {
+	// Validate one of config or activity log was enabled; otherwise error out
+	if !args.ActivityLog && !args.Config {
+		return errors.New("audit log or config integration must be enabled")
+	}
+
+	// Validate that active directory settings are correct
+	if !args.AdIntegration {
+		if args.AdApplicationId == "" || args.AdServicePrincipalId == "" || args.AdApplicationPassword == "" {
+			return errors.New("Active directory details must be set")
+		}
+	}
+
+	// Validate the Mangement Group
+	if args.ManagementGroup && args.ManagementGroupId == "" {
+		return errors.New("When Group Management is enabled, then Group Id must be configured")
+	}
+
+	// Validate Storage Account
+	if args.ExistingStorageAccount {
+		if args.StorageAccountName == "" || args.StorageAccountResourceGroup == "" {
+			return errors.New("When using existing storage account, storage account details must be configured")
+		}
+	}
+
+	return nil
+}
+
+type AzureTerraformModifier func(c *GenerateAzureTfConfigurationArgs)
+
+// NewTerraform returns an instance of the GenerateAzureTfConfigurationArgs struct with the provided enabled
+// settings (config/activity log).
+//
+// Note: Additional configuration details may be set using modifiers of the AzureTerraformModifier type
+//
+func NewTerraform(enableConfig bool, enableActivityLog bool, mods ...AzureTerraformModifier) *GenerateAzureTfConfigurationArgs {
```

Comment:
When I run this without any modifiers (i.e., `azure.NewTerraform(true, true).Generate()`), I get: 

```
invalid inputs: Active directory details must be set
exit status 1
```

I believe we should make the AD settings a required input for the constructor since it cannot work without this value being set.

Reply:
I think this is correct, the user can choose to create an Active Directory application, this adds hcl into the terraform to do so, if the do not use the create then they must supply the active directory attributes for an already existing application

Reply:
I guess my confusion is that this is a constructor, and AD settings are required for this code to work (if I'm understanding correctly); so therefore I would think the AD settings would be a required 3rd property?  The whole point (IMO) of the constructor is to make using this large struct easier, and I would think the constructor should guide the developer to a working configuration.

Reply:
Made this a constructor now